device/core/test/_device_client_test.js

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var assert = require('chai').assert;
+var sinon = require('sinon');
+var stream = require('stream');
+var EventEmitter = require('events').EventEmitter;
+var FakeTransport = require('./fake_transport.js');
+var Message = require('azure-iot-common').Message;
+var errors = require('azure-iot-common').errors;
+var results = require('azure-iot-common').results;
+var X509AuthenticationProvider = require('../lib/x509_authentication_provider').X509AuthenticationProvider;
+var Client = require('../lib/device_client').Client;
+
+describe('Device Client', function () {
+  var sharedKeyConnectionString = 'HostName=host;DeviceId=id;SharedAccessKey=key';
+  var sharedAccessSignature = '"SharedAccessSignature sr=hubName.azure-devices.net/devices/deviceId&sig=s1gn4tur3&se=1454204843"';
+
+  describe('#fromConnectionString', function () {
+    /*Tests_SRS_NODE_DEVICE_CLIENT_05_006: [The fromConnectionString method shall return a new instance of the InternalClient object, as by a call to new InternalClient(new Transport(...)).]*/
```

Comment:
>InternalClient [](start = 154, length = 14)

Similar comment as in the ModuleClient factory methods.  the requirements refer to InternalClient still.